### ISSUE TITLE
Objects without modifier rendered slowly than with modifier

### ISSUE THREAD

**Issue by youle31** [UPBGE/blender#139]
BF Blender with modifier: http://www.pasteall.org/pic/show.php?id=104280
BF Blender without modifier: http://www.pasteall.org/pic/show.php?id=104281
UPBGE with modifier: http://www.pasteall.org/pic/show.php?id=104282
UPBGE without modifier: http://www.pasteall.org/pic/show.php?id=104283

test files: 
- with modifier: http://www.pasteall.org/blend/42406
- without modifier: https://drive.google.com/file/d/0B3GouQIyoCmrdHMweTV2VTJiSkk/view?usp=sharing

**Comment by panzergame** [comment]
Did you test by forcing DA+DL or VBO ?

**Comment by youle31** [comment]
I'm quite sure this is this commit: https://github.com/UPBGE/blender/commit/69ac771ede7a5679b4f3a2c0d89c1493f38231fa that is the problem but I can't confirm because I can't compile. I'll make tests with quads to see

**Comment by panzergame** [comment]
in UPBGE i have a equal result if i use VBO instead of VA+DL, but yes VA are slower…

**Comment by panzergame** [comment]
With a fixed framerate UPBGE is slower of 6% (0.29ms vs 0.31ms) is a very slight difference but it can be normal due to the use of DL in UPBGE. But with VBO it's a nice reverse case 12% faster for UPBGE (0.35ms vs 0.40ms). Well, as the DL+VA will be deprecated i'm not very not afraid of these 6%, but you high-lined an issue: VA+DL are currently the best render mode for big meshes.

For modifier, i tested with VA+DL. It's faster only in BF BGE of 13% (0.25ms vs 0.29ms), but in UPBGE it's slower of 40% (0.31ms vs 0.52ms). Here too it's normal because previously in BF BGE we was doing a workaround to use DL for modifier, this workaround was totally unsafe and was removed. But unfortunately VBO are slower of  15% in UPBGE (0.45ms vs 0.54).

To summarize, in the case of big meshes: UPBGE in VA+DL is a bit slower than BFBGE in regular render and modifiers, but in VBO UPBGE is much faster than BFBGE for regular but not for modifiers.

Anyway i think these test are not very representative because it depends a lot of the GPU and it's pretty rare to render 800000 faces in the same view.

**Comment by youle31** [comment]
I don't understand... When i apply modifiers, i loose 220 fps (660 -440) in upbge. You Can test using the bpy script in the other thread. It's a considérable loose of fps in va plus dl... It's the exact opposite of what happens in bf bge where render is slower with modifier than without modifier. I don't understand also why use "fixed frame rate". And why look at render time instead of fps as both are linked.......

**Comment by panzergame** [comment]
Using a fixed framerate is to avoid stree the GPU and get incorrect result. The best example is GI : it's 70% faster with a small mesh but it can be 200% slower if you use to many vertices. In your test i will considerate that the GPU is not stressed.

I still have opposite result than you:
                | BF BGE | UPBGE |
modifiers |     +        |      -       |
regular    |      -        |      +      |

**Comment by youle31** [comment]
okok... My computer is a bit weird so. Benoit Bolsee told me that he had better performances with his patch than without and I have the opposite and here too... Thanks for your tests. and good night!

**Comment by youle31** [comment]
I close this because it seems we have not the same behavior on all computers... weird.
